px4_ros2_cpp/include/px4_ros2/navigation/experimental/local_navigation_interface.hpp

```diff
@@ -0,0 +1,145 @@
+ * SPDX-License-Identifier: BSD-3-Clause
+ ****************************************************************************/
+
+#pragma once
+
+#include <optional>
+#include <rclcpp/rclcpp.hpp>
+#include <Eigen/Eigen>
+
+#include <px4_msgs/msg/vehicle_odometry.hpp>
+#include <px4_ros2/navigation/experimental/navigation_interface_base.hpp>
+
+namespace px4_ros2
+{
+
+using AuxLocalPosition = px4_msgs::msg::VehicleOdometry;
+
+enum class PoseFrame
+{
+  Unknown,
+  NED,
+  FRD
+};
+
+enum class VelocityFrame
+{
+  Unknown,
+  NED,
+  FRD,
+  BodyFRD
+};
+
+constexpr inline uint8_t poseFrameToMessageFrame(px4_ros2::PoseFrame frame) noexcept
+{
+  switch (frame) {
+    case PoseFrame::Unknown: return AuxLocalPosition::POSE_FRAME_UNKNOWN;
+
+    case PoseFrame::NED: return AuxLocalPosition::POSE_FRAME_NED;
+
+    case PoseFrame::FRD: return AuxLocalPosition::POSE_FRAME_FRD;
+  }
+
+  return AuxLocalPosition::POSE_FRAME_UNKNOWN;
+}
+
+constexpr inline uint8_t velocityFrameToMessageFrame(px4_ros2::VelocityFrame frame) noexcept
+{
+  switch (frame) {
+    case VelocityFrame::Unknown: return AuxLocalPosition::VELOCITY_FRAME_UNKNOWN;
+
+    case VelocityFrame::NED: return AuxLocalPosition::VELOCITY_FRAME_NED;
+
+    case VelocityFrame::FRD: return AuxLocalPosition::VELOCITY_FRAME_FRD;
+
+    case VelocityFrame::BodyFRD: return AuxLocalPosition::VELOCITY_FRAME_BODY_FRD;
+  }
+
+  return AuxLocalPosition::POSE_FRAME_UNKNOWN;
+}
+
+/**
+ * @struct LocalPositionEstimate
+ * @brief Represents a local position estimate to be passed to `LocalNavigationInterface::update`.
+ *
+ * This struct holds information about the local position estimate, including: the timestamp of the sample, vertical and horizontal postion and velocity, attitude, and their associated variances.
+ * @see LocalNavigationInterface::update
+ */
+struct LocalPositionEstimate
+{
+  /** @brief Timestamp of the sample. */
+  rclcpp::Time timestamp_sample {};
+
+  /** @brief Position in the xy-plane. */
+  std::optional<Eigen::Vector2f> position_xy {std::nullopt};
+  /** @brief Variance of position error in the xy-plane. */
+  std::optional<Eigen::Vector2f> position_xy_variance {std::nullopt};
+  /** @brief Position in the z-axis. */
+  std::optional<float> position_z {std::nullopt};
+  /** @brief Variance of position error in the z-axis. */
+  std::optional<float> position_z_variance {std::nullopt};
+
+  /** @brief Velocity in the xy-plane. */
+  std::optional<Eigen::Vector2f> velocity_xy {std::nullopt};
+  /** @brief Variance of velocity error in the xy-plane. */
+  std::optional<Eigen::Vector2f> velocity_xy_variance {std::nullopt};
+  /** @brief Velocity in the z-axis. */
+  std::optional<float> velocity_z {std::nullopt};
+  /** @brief Variance of velocity error in the z-axis. */
+  std::optional<float> velocity_z_variance {std::nullopt};
+
+  /** @brief Attitude quaternion. */
+  std::optional<Eigen::Quaternionf> attitude_quaternion {std::nullopt};
+  /** @brief Variance of attitude error. */
```

Comment:
```suggestion
  /** @brief Variance of attitude in body frame. */
```
It's also possible to define the attitude variance in the local frame, so we should make it explicit.